In [1]:
#Import the libraries


In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [3]:
#To read the dataset

In [4]:
df = pd.read_csv('StudentsPerformance.csv')

In [5]:
df.head(10)

gender race/ethnicity parental level of education         lunch  \
0  female        group C            some high school  free/reduced   
1  female        group B                 high school  free/reduced   
2  female        group B            some high school  free/reduced   
3  female        group B                some college      standard   
4  female        group C                some college  free/reduced   
5  female        group B                 high school  free/reduced   
6  female        group B            some high school  free/reduced   
7  female        group D          associate's degree  free/reduced   
8  female        group D            some high school  free/reduced   
9  female        group C                 high school      standard   

  test preparation course  math score  reading score  writing score  
0                    none           0             17             10  
1                    none           8             24             23  
2                    none          18             32             28  
3                    none          11             38             32  
4                    none          22             39             33  
5               completed          23             44             36  
6                    none          24             38             27  
7                    none          26             31             38  
8                    none          19             34             32  
9                    none          29             29             30

In [6]:
#To check the null value

In [7]:
df.isnull().sum()

gender                         0
race/ethnicity                 0
parental level of education    0
lunch                          0
test preparation course        0
math score                     0
reading score                  0
writing score                  0
dtype: int64

# 1) Find out how many males and females participated in the test.

In [8]:
df['gender'].value_counts()['male']

482

In [9]:
df['gender'].value_counts()['female']

518

482 males and 518 Females are parcipated in the test.

# OR

In [10]:
df['gender'].value_counts()

female    518
male      482
Name: gender, dtype: int64

# 2. What do you think about the students' parental level of education?

In [11]:
df['parental level of education'].unique()

array(['some high school', 'high school', 'some college',
       "associate's degree", "bachelor's degree", "master's degree"],
      dtype=object)

In [13]:
df['parental level of education'].value_counts()

some college          226
associate's degree    222
high school           196
some high school      179
bachelor's degree     118
master's degree        59
Name: parental level of education, dtype: int64

In [15]:
#INSIGHTS

There are 6 parental level of education.Most of the student's parental level of education is some college.some of them have associate's degree.Only 59 students have master's degree.

# 3. Who scores the most on average for math, reading and writing based on
● Gender
● Test preparation course

In [17]:
df[['gender','math score','reading score','writing score']].groupby('gender').mean()

math score  reading score  writing score
gender                                          
female   63.376448      72.590734      72.467181
male     68.821577      65.545643      63.446058

In [16]:
#INSIGHTS

Males have higher math score than females,whereas Females have higher scores in reading and writing than males.

In [18]:
df[['test preparation course','math score','reading score','writing score']].groupby('test preparation course').mean()

math score  reading score  writing score
test preparation course                                          
completed                  69.96648      74.175978      74.684358
none                       63.78972      66.417445      64.457944

In [ ]:
#INSIGHTS

Students who completed the Test preparation course have higher scores in math,reading,writing than those who haven't taken the course.

# 4. What do you think about the scoring variation for math, reading and writing
based on
● Gender
● Test preparation course

In [19]:
df[['gender','math score','reading score','writing score']].groupby('gender').var()

math score  reading score  writing score
gender                                          
female  256.958593     207.677438     220.369327
male    211.889097     200.211010     202.413924

In [ ]:
#INSIGHTS

When we looking on the scoring variation for math,reading,writing based on gender,Females scoring higher in these three catogories than males.

In [20]:
df[['test preparation course','math score','reading score','writing score']].groupby('test preparation course').var()

math score  reading score  writing score
test preparation course                                          
completed                210.884027     183.265864     175.202612
none                     246.668662     213.419851     226.251739

In [21]:
#INSIGHTS

Whereas in the case of scoring variation on the basis of test preparation course,the students who not completed the test preparation course scoring higher in math,reading,writing than who have taken the course.

# 5. The management needs your help to give bonus points to the top 25% of
students based on their maths score, so how will you help the management
to achieve this.

In [28]:
df['math score'].describe()

count    1000.000000
mean       66.001000
std        15.569567
min         0.000000
25%        57.000000
50%        66.000000
75%        77.000000
max       100.000000
Name: math score, dtype: float64

In [27]:
df['math score'].quantile(0.75)

77.0

In [26]:
df.sort_values(by = 'math score',ascending = False).head(250)

gender race/ethnicity parental level of education         lunch  \
999    male        group E           bachelor's degree      standard   
996    male        group A                some college      standard   
515  female        group E                some college      standard   
517  female        group E          associate's degree      standard   
516  female        group E           bachelor's degree      standard   
..      ...            ...                         ...           ...   
856    male        group E            some high school      standard   
855    male        group E          associate's degree  free/reduced   
854    male        group D            some high school      standard   
853    male        group D          associate's degree  free/reduced   
852    male        group D                some college  free/reduced   

    test preparation course  math score  reading score  writing score  
999               completed         100            100            100  
996               completed         100             96             86  
515                    none         100             92             97  
517                    none         100            100            100  
516                    none         100            100            100  
..                      ...         ...            ...            ...  
856               completed          77             76             77  
855               completed          77             69             68  
854               completed          77             68             69  
853                    none          77             78             73  
852                    none          77             62             64  

[250 rows x 8 columns]

#INSIGHTS
The students who score 77 and more than 77 comprises the top 25% marks in maths.The above students are eligible for the bonus point based on their math score.